yew-ui/src/model/media_devices/media_device_access.rs

```diff
@@ -0,0 +1,62 @@
+use gloo_utils::window;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use wasm_bindgen::prelude::*;
+use wasm_bindgen_futures::JsFuture;
+use web_sys::MediaStreamConstraints;
+use yew::prelude::Callback;
+
+pub struct MediaDeviceAccess {
+    granted: Arc<AtomicBool>,
+    pub on_granted: Callback<()>,
+    pub on_denied: Callback<()>,
+}
+
+impl MediaDeviceAccess {
+    pub fn new() -> Self {
```

Comment:
@ronen if you pass on_granted and on_denied to the new contructor, then the callbacks do not have to be public.

Reply:
Yes, but then you have an API requiring a bunch of positional parameters of the same type, which IMHO doesn't make for a great extensible API.   I'd say better would be to use the "options struct" pattern such as used elsewhere in the code base (or builder pattern), but I was getting too lazy.   (Guess where I fall on the endless should-Rust-have-named-parameters debate? :)

In any case, I'm thinking that once #74 is done, it'd be worth taking a pass over the entire API for the client-side engine, and come up with something clean & consistent (and not succumb to laziness :)

But if you feel strongly I can change it to something else now.

Reply:
I do not feel strongly about it, fixing the callback on the other comment will do 👍